client/jetpack-cloud/sections/agency-dashboard/sites-overview/site-welcome-banner/index.tsx

```diff
@@ -0,0 +1,82 @@
+import { useTranslate } from 'i18n-calypso';
+import { ReactElement, useCallback, useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import tipIcon from 'calypso/assets/images/jetpack/tip-icon.svg';
+import Banner from 'calypso/components/banner';
+import {
+	JETPACK_DASHBOARD_WELCOME_BANNER_PREFERENCE,
+	JETPACK_DASHBOARD_WELCOME_BANNER_PREFERENCE_HOME_PAGE as homePagePreferenceName,
+	getJetpackDashboardWelcomeBannerPreference as getPreference,
+} from 'calypso/state/partner-portal/agency-dashboard/selectors';
+import { savePreference } from 'calypso/state/preferences/actions';
+import type { Preference, PreferenceType } from '../types';
+
+export default function SiteWelcomeBanner( {
+	isDashboardView,
+	bannerKey,
+}: {
+	isDashboardView?: boolean;
+	bannerKey?: string;
+} ): ReactElement | null {
+	const translate = useTranslate();
+	const dispatch = useDispatch();
+
+	const preferenceName = isDashboardView
+		? homePagePreferenceName
+		: `${ JETPACK_DASHBOARD_WELCOME_BANNER_PREFERENCE }-${ bannerKey }`;
+	const preference: Preference = useSelector( ( state ) => getPreference( state, preferenceName ) );
+	const homePagePreference: Preference = useSelector( ( state ) =>
```

Comment:
You don't need to define the type here because TypeScript infers the type automatically here from return type of `getPreference`

<img width="967" alt="Screenshot 2022-05-24 at 5 11 20 PM" src="https://user-images.githubusercontent.com/18226415/170026862-4d96f2fc-b888-4335-8a76-e3f80bdd8376.png">


```suggestion
	const preference = useSelector( ( state ) => getPreference( state, preferenceName ) );
	const homePagePreference = useSelector( ( state ) =>
```